nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/PutBoxFile.java

```diff
@@ -0,0 +1,407 @@
+                }
+
+                if (uploadedFileInfo == null) {
+                    if (size > chunkUploadThreshold) {
+                        uploadedFileInfo = parentFolder.uploadLargeFile(rawIn, filename, size);
+                    } else {
+                        uploadedFileInfo = parentFolder.uploadFile(rawIn, filename);
+                    }
+                }
+            } catch (BoxAPIResponseException e) {
+                if (e.getResponseCode() == CONFLICT_RESPONSE_CODE) {
+                    handleConflict(conflictResolution, filename, parentFolder, e);
+                } else {
+                    throw e;
+                }
+            }
+
+            if (uploadedFileInfo != null) {
+                final Map<String, String> attributes = BoxFileUtils.createAttributeMap(uploadedFileInfo);
+                final String url = BOX_URL + uploadedFileInfo.getID();
+                flowFile = session.putAllAttributes(flowFile, attributes);
+                final long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                session.getProvenanceReporter().send(flowFile, url, transferMillis);
+            }
+
+            session.transfer(flowFile, REL_SUCCESS);
+        } catch (BoxAPIResponseException e) {
+            getLogger().error("Upload failed: File [{}] Folder [{}] Response Code [{}]", filename, fullPath, e.getResponseCode(), e);
+            handleExpectedError(session, flowFile, e);
+        } catch (Exception e) {
+            getLogger().error("Upload failed: File [{}], Folder [{}]", filename, fullPath, e);
+            handleUnexpectedError(session, flowFile, e);
+        }
+    }
+
+    private BoxFolder getOrCreateDirectParentFolder(ProcessContext context, FlowFile flowFile ) {
+        final String subfolderPath = context.getProperty(SUBFOLDER_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final boolean createFolder = context.getProperty(CREATE_SUBFOLDER).asBoolean();
+        final String folderId = context.getProperty(FOLDER_ID).evaluateAttributeExpressions(flowFile).getValue();
+        BoxFolder parentFolder = getFolderById(folderId);
+
+        if (subfolderPath != null) {
+            final Queue<String> subFolderNames = getSubFolderNames(subfolderPath);
+            parentFolder = getFolder(getOrCreateSubfolders(subFolderNames, parentFolder, createFolder).getID());
+        }
+
+        return parentFolder;
+    }
+
+    private BoxFile.Info replaceExistingBoxFile(BoxFolder parentFolder, String filename, final InputStream rawIn, final long size, final long chunkUploadThreshold)
+            throws IOException, InterruptedException {
+        final Optional<BoxFile.Info> existingBoxFileInfo = getFileByName(filename, parentFolder);
+        if (existingBoxFileInfo.isPresent()) {
+            final BoxFile existingBoxFile = new BoxFile(boxAPIConnection, existingBoxFileInfo.get().getID());
```

Comment:
Using `BoxFile.Info.getResource()` would be more API compliant I think:
```suggestion
            final BoxFile existingBoxFile = existingBoxFileInfo.get().getResource();
```
Though it requires more mocking in some tests:
```
        when(mockSubfolder1Info.getResource()).thenReturn(mockSubfolder1);
        when(mockSubfolder2Info.getResource()).thenReturn(mockSubfolder2);
```